### PR TITLE
Enhancing IP Validation logic for ExternalAccess(for NFS vol) IP

### PR DESCRIPTION
**Pull request by adarsh-dell**
# Description
Enhancing IP Validation logic for ExternalAccess(for NFS vol) IP

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/689 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran make test for testing if any existing UT is getting broken or not. 
image

- [x] Tested with multiple valid and Invalid IP's combinations and got expected results.

### CODE REVIEW

File: pkg/common/common.go [hunk 1]
```diff
@@ -203,18 +203,35 @@ func parseMask(ipaddr string) (mask string, err error) {
...
 // A return value of nil indicates no match
 func GetIPListWithMaskFromString(input string) (string, error) {
-	re := regexp.MustCompile(`^([01]?\d\d?|2[0-4]\d|25[0-5])(?:\.(?:[01]?\d\d?|2[0-4]\d|25[0-5])){3}(?:/[0-2]\d|/3[0-2])?$`)
-	validated := re.FindAllString(input, 1)
-	if validated != nil {
-		mask, err := parseMask(validated[0])
+	// Split the IP address and subnet mask if present
+	parts := strings.Split(input, "/")
+	ip := parts[0]
+	result := net.ParseIP(parts[0])
```
Comment (niranjan-n1) [inline pkg/common/common.go:209]: Can variable 'ip' be used here parts[0]

File: pkg/common/common.go [hunk 2]
```diff
@@ -203,18 +203,35 @@ func parseMask(ipaddr string) (mask string, err error) {
...
+	if result == nil {
+		return "", errors.New("doesn't seem to be a valid IP")
+	}
+	if len(parts) > 1 {
```
Comment (niranjan-n1) [inline pkg/common/common.go:213]: can this check be done before net.ParseIP(parts[0]) ??

File: pkg/common/common.go (hunk 1 above)
Comment (adarsh-dell) [reply pkg/common/common.go]: Done, Thanks.

File: pkg/common/common.go (hunk 2 above)
Comment (adarsh-dell) [reply pkg/common/common.go:213]: Based on the use case of this environment variable in the customer's production setup, it is more common for them to pass plain IP addresses without subnet masks, which increases the likelihood of entering invalid IP addresses wrt invalid mask ID or anything else. So I thought of putting as a second check.

**Review by niranjan-n1 — APPROVED** [review]

**Review by suryagupta4 — APPROVED** [review]

**Pull request merged**